src/new/sendTransaction.js

```diff
@@ -287,21 +292,24 @@ class SendTransaction extends EventEmitter {
    * @inner
    */
   async runFromMining(until = null) {
-    const promise = new Promise((resolve, reject) => {
+    const promise = new Promise(async (resolve, reject) => {
```

Comment:
Do we really need a Promise wrapping this code here?

Why not just this (which I think works the same, but didn't test to confirm)?

```ts
 async runFromMining(until = null) {
    try {
      // This will await until mine tx is fully completed
      // mineTx method returns a promise that resolves when
      // mining succeeds or rejects when there is an error
      const mineData = await this.mineTx();
      this.transaction.parents = mineData.parents;
      this.transaction.timestamp = mineData.timestamp;
      this.transaction.nonce = mineData.nonce;
      this.transaction.weight = mineData.weight;

      if (until === 'mine-tx') {
        return this.transaction;
      }

      const pushedTx = await this.handlePushTx();
      return this.transaction;
    } catch (err) {
      if (err instanceof WalletError) {
        this.emit('send-error', err.message);
      }
      throw err;
    }
  }
```

Reply:
I agree it's much better. Refactor done here https://github.com/HathorNetwork/hathor-wallet-lib/pull/276/commits/7b6165efb359a999abd7869a5ca3173d3d99a232. Can you see again please?